src/services/chat/repository.ts

```diff
@@ -1,117 +1,125 @@
-import { In } from 'typeorm';
+import { EntityManager, In } from 'typeorm';
 
 import { ResultOf } from '@graasp/sdk';
 
-import { AppDataSource } from '../../plugins/datasource';
-import { MemberIdentifierNotFound } from '../itemLogin/errors';
+import { MutableRepository } from '../../repositories/MutableRepository';
+import { DEFAULT_PRIMARY_KEY } from '../../repositories/const';
+import { DeleteException } from '../../repositories/errors';
+import { GetForItem, GetForItems, GetForMemberExport } from '../../repositories/interfaces';
 import { Member } from '../member/entities/member';
 import { messageSchema } from '../member/plugins/export-data/schemas/schemas';
 import { schemaToSelectMapper } from '../member/plugins/export-data/utils/selection.utils';
 import { mapById } from '../utils';
 import { ChatMessage } from './chatMessage';
-import { ChatMessageNotFound } from './errors';
 
-export const ChatMessageRepository = AppDataSource.getRepository(ChatMessage).extend({
+// This type simplify the readability of the Repository's implements.
+type IChatMessageRepository<T extends ChatMessage> = GetForItem<T> &
+  GetForItems<T> &
+  GetForMemberExport<T>;
+
+type ChatMessageUpdateBody = Partial<ChatMessage>;
+type ChatMessageCreateBody = { itemId: string; creator: Member; body: string };
+
+export class ChatMessageRepository
+  extends MutableRepository<ChatMessage, ChatMessageUpdateBody>
+  implements IChatMessageRepository<ChatMessage>
```

Comment:
I think it is a really interesting way to keep the naming of our functions consistant.

But, we need to keep in mind that in the future, we're gonna use tsyringe and will get our Repository from the class name, and will rarely use this interface.. To be honest, I feel like we're not using interface how they're intented to be used here.

Have you considered to use ES-Lint or any equivalent ? I just saw that : https://typescript-eslint.io/rules/naming-convention/, could this apply to our issue ?

By the way, I think we should keep it as it is, and find a solution on another step of this refactoring.

Reply:
On my point of view, using interfaces allow us to describe the contract between the repository and the client code (code that are using the repository) as well as allowing consistency in the signature of the functions. Each time a class implements the interface (GetAll) for example, the class guarantees the existence of the method getAll() which will return an array of T.

I didn't check in depth, but I'm not sure using ES-Lint allows us to enforce naming functions and return types, and even if it's the case, I'm wondering if it's the best solution. Maybe I'm wrong, but in my sense, using interfaces give us the possibility to enforce a consistent method signature per action.

I understand one potential pitfall which is the greater number of interfaces, but I don't think it's a big issue.

What do you think @Defferrard and @spaenleh @MatveyK @pyphilia ? I'm always open to changes ;)